modules/mastodon/index.js

```diff
@@ -0,0 +1,320 @@
+/*
+ * This module prints ActivityPub Note and Person objects based on a URL. Its
+ * primary use is to print information about permalinks on Mastodon instances,
+ * hence the module name, but could theoretically be used on any endpoint that
+ * supports those ActivityPub object types.
+ *
+ * Because of the focus on Mastodon instances (on a sufficiently recent
+ * version), assumptions are made about how to operate. This module is not
+ * expected to be a fully-fledged ActivityPub client. Issues with handling URLs
+ * on Mastodon 1.6.0 and later are considered bugs. Other issues will be
+ * handled on a case-by-case basis.
+ *
+ * Mastodon instances can be on any hostname. Instance autodiscovery is
+ * somewhat irresponsible as it would cause a request to be made every time a
+ * URL is posted to a channel.
+ *
+ * Instances are thus discovered from other instances organically. A small list
+ * of instances is included in default_instances.json, which is used in a
+ * default configuration. When a toot is fetched, this bot also fetches
+ * /api/v1/timelines/public and learns new instances from that.
+ *
+ * In case an instance isn't already known, users can share a toot URL with the
+ * !masto command, which will force the bot to do an API lookup.
+ *
+ * Bot administrators can block hostnames from being used as instances by
+ * adding them to the "blocked" key in this module's config.json.
+ */
+
+var arrayUniq = require('array-uniq')
+  , Cacheman = require('cacheman')
+  , contentType = require('content-type')
+  , execSync = require('child_process').execSync
+  , htmlToTextMod = require('html-to-text')
+  , robotsParser = require('robots-parser')
+  , robotsTxtCache = new Cacheman()
+  , sprintf = require('extsprintf').sprintf
+  , util = require('util')
+  , urlMod = require('url');
+
+/* variables defined in module.exports.init */
+var bot = null
+  , log = null
+  , moduleConfig = null
+  , request = null;
+
+function robotRequest(options, fn) {
+  var url = options.url;
+  var userAgent = moduleConfig != null && moduleConfig.userAgent || 'EuIrcBot';
```

Comment:
I think a ternary is a more idiomatic way to do a one-liner like this? If a ternary can ever be idiomatic? I feel like I stumbled into a little bit of bash here.